papers/extract.py

```diff
@@ -22,21 +20,55 @@
 # PDF parsing / crossref requests
 # ===============================
 
+def createrandname(path, ext):
+    import random
+    import string
+    check = False
+    while check == False:
+        rand = "".join(
+            random.sample(string.ascii_lowercase + string.digits, 20)
+        ) + ext
+        check = True if not os.path.exists(path + rand) else False
+    return rand
+
+
 def readpdf(pdf, first=None, last=None, keeptxt=False):
-    txtfile = pdf.replace('.pdf','.txt')
+    import shutil
+    txtfile = pdf.replace('.pdf', '.txt')
+    imgfile = pdf.replace('.pdf', '.png')
     # txtfile = os.path.join(os.path.dirname(pdf), pdf.replace('.pdf','.txt'))
-    if True: #not os.path.exists(txtfile):
+    if True:  # not os.path.exists(txtfile):
         # logger.info(' '.join(['pdftotext','"'+pdf+'"', '"'+txtfile+'"']))
-        cmd = ['pdftotext']
-        if first is not None: cmd.extend(['-f',str(first)])
-        if last is not None: cmd.extend(['-l',str(last)])
+        if os.path.isfile(pdf):
+            path, ext = os.path.splitext(pdf)
+            path = "/".join(path.split("/")[:-1])
+            rand = createrandname(path, ext)
```

Comment:
```suggestion
            import tempfile
            rand = tempfile.mkstemp(suffix=ext)¶
```

Reply:
Right, here we are missing something. It is a bit old now, but if I recall correctly, `keeptxt` was meant to speedup some of the testing. In that case, we do not want random naming. But I also see that it is never used (it was commented out), so I suggest to just use temporary files with tempfile module, so that we don't even need to delete them afterward.

Reply:
Cool, I wasn't aware of this! Thanks.

Reply:
That is doable but for someone like me with 600 pdf's imagine not deleting those extra files in the `/tmp` and I also rarely restart my pc so the '/tmp' doesn't get cleaned up. Wanna have a look on the last PR and decide if u wanna proceed with keeping them or removing them. The last PR has meaningful var names and it cleans up the extra files.

Reply:
ok i'll take a look at the last PR, but to answer now:
- it is still possible to remove them, even in the /tmp directory
- tempfile is the way to go in any case, see for instance the example I provided, it does not use /tmp